src/modules/roc_audio/profiler.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_audio/profiler.h
+//! @brief Profiler.
+
+#ifndef ROC_AUDIO_PROFILER_H_
+#define ROC_AUDIO_PROFILER_H_
+
+#include "roc_audio/frame.h"
+#include "roc_core/array.h"
+#include "roc_core/iallocator.h"
+#include "roc_core/list.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/rate_limiter.h"
+#include "roc_core/refcnt.h"
+#include "roc_core/time.h"
+#include "roc_packet/units.h"
+
+namespace roc {
+namespace audio {
+
+namespace {
+
+const core::nanoseconds_t LogInterval = core::Second;
+
+} // namespace
+
+//! Profiler
+class Profiler : public core::NonCopyable<> {
+public:
+    //! Initialization.
+    Profiler(core::IAllocator& allocator,
+             packet::channel_mask_t channels,
+             size_t sample_rate);
+
+    //! Init audio frame.
+    void begin_frame(size_t frame_size);
+
+    //! Profile frame speed
+    void end_frame(size_t frame_size,
+                   core::nanoseconds_t elapsed,
+                   const core::nanoseconds_t interval = LogInterval);
```

Comment:
Can we make interval the argument of constructor instead of end_frame()? It would be strange to have a different interval for different frames. It would also be strange to have different intervals for log reporting and for calculations.

Also, we usually try to avoid default arguments.